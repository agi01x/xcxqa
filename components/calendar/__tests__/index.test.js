import React from 'react';
import Moment from 'moment';
import { mount } from 'enzyme';
import MockDate from 'mockdate';
import Calendar from '..';

describe('Calendar', () => {
  it('Calendar should be selectable', () => {
    const onSelect = jest.fn();
    const wrapper = mount(
      <Calendar onSelect={onSelect} />
    );
    wrapper.find('.ant-fullcalendar-cell').at(0).simulate('click');
    expect(onSelect).toBeCalledWith(expect.anything());
    const value = onSelect.mock.calls[0][0];
    expect(Moment.isMoment(value)).toBe(true);
  });

  it('only Valid range should be selectable', () => {
    const onSelect = jest.fn();
    const validRange = [Moment('2018-02-02'), Moment('2018-02-18')];
    const wrapper = mount(
      <Calendar onSelect={onSelect} validRange={validRange} defaultValue={Moment('2018-02-02')} />
    );
    wrapper.find('[title="February 1, 2018"]').at(0).simulate('click');
    wrapper.find('[title="February 2, 2018"]').at(0).simulate('click');
    expect(onSelect.mock.calls.length).toBe(1);
  });

  it('dates other than in valid range should be disabled', () => {
    const onSelect = jest.fn();
    const validRange = [Moment('2018-02-02'), Moment('2018-02-18')];
    const wrapper = mount(
      <Calendar onSelect={onSelect} validRange={validRange} defaultValue={Moment('2018-02-02')} />
    );
    wrapper.find('[title="February 20, 2018"]').at(0).simulate('click');
    const elem = wrapper.find('[title="February 20, 2018"]').hasClass('ant-fullcalendar-disabled-cell');
    expect(elem).toEqual(true);
    expect(onSelect.mock.calls.length).toBe(0);
  });

  it('months other than in valid range should be disabled', () => {
    const onSelect = jest.fn();
    const validRange = [Moment('2018-02-02'), Moment('2018-05-18')];
    const wrapper = mount(
      <Calendar onSelect={onSelect} validRange={validRange} defaultValue={Moment('2018-02-02')} mode="year" />
    );
    expect(wrapper.find('[title="Jan"]').at(0).hasClass('ant-fullcalendar-month-panel-cell-disabled')).toBe(true);
    expect(wrapper.find('[title="Feb"]').at(0).hasClass('ant-fullcalendar-month-panel-cell-disabled')).toBe(false);
    expect(wrapper.find('[title="Jun"]').at(0).hasClass('ant-fullcalendar-month-panel-cell-disabled')).toBe(true);
    wrapper.find('[title="Jan"]').at(0).simulate('click');
    wrapper.find('[title="Mar"]').at(0).simulate('click');
    expect(onSelect.mock.calls.length).toBe(1);
  });

  it('months other than in valid range should not be shown in header', () => {
    const validRange = [Moment('2017-02-02'), Moment('2018-05-18')];
    const wrapper = mount(
      <Calendar validRange={validRange} />
    );
    wrapper.find('.ant-fullcalendar-year-select').hostNodes().simulate('click');
    wrapper.find('.ant-select-dropdown-menu-item').first().simulate('click');
    wrapper.find('.ant-fullcalendar-month-select').hostNodes().simulate('click');
    // 2 years and 11 months
    expect(wrapper.find('.ant-select-dropdown-menu-item').length).toBe(13);
  });

  it('getDateRange should returns a disabledDate function', () => {
    const validRange = [Moment('2018-02-02'), Moment('2018-05-18')];
    const wrapper = mount(
      <Calendar validRange={validRange} defaultValue={Moment('2018-02-02')} />
    );
    const instance = wrapper.instance();
    const disabledDate = instance.getDateRange(validRange);
    expect(disabledDate(Moment('2018-06-02'))).toBe(true);
    expect(disabledDate(Moment('2018-04-02'))).toBe(false);
  });

  it('Calendar should change mode by prop', () => {
    const monthMode = 'month';
    const yearMode = 'year';
    const wrapper = mount(
      <Calendar />
    );
    expect(wrapper.state().mode).toEqual(monthMode);
    wrapper.setProps({ mode: 'year' });
    expect(wrapper.state().mode).toEqual(yearMode);
  });

  it('Calendar should switch mode', () => {
    const monthMode = 'month';
    const yearMode = 'year';
    const onPanelChangeStub = jest.fn();
    const wrapper = mount(
      <Calendar mode={yearMode} onPanelChange={onPanelChangeStub} />
    );
    expect(wrapper.state().mode).toEqual(yearMode);
    wrapper.instance().setType('date');
    expect(wrapper.state().mode).toEqual(monthMode);
    expect(onPanelChangeStub).toHaveBeenCalledTimes(1);
  });

  it('Calendar should support locale', () => {
    MockDate.set(Moment('2018-10-19'));
    // eslint-disable-next-line
    const zhCN = require('../locale/zh_CN').default;
    const wrapper = mount(
      <Calendar locale={zhCN} />
    );
    expect(wrapper.render()).toMatchSnapshot();
    MockDate.reset();
  });
});
