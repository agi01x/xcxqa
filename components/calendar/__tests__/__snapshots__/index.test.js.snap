// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Calendar should support locale 1`] = `
<div
  class=" ant-fullcalendar-fullscreen"
>
  <div
    class="ant-fullcalendar-header"
  >
    <div
      class="ant-fullcalendar-year-select ant-select ant-select-enabled"
    >
      <div
        aria-autocomplete="list"
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-select-selection
            ant-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="ant-select-selection__rendered"
        >
          <div
            class="ant-select-selection-selected-value"
            style="display: block; opacity: 1;"
            title="2018年"
          >
            2018年
          </div>
        </div>
        <span
          class="ant-select-arrow"
          unselectable="on"
        >
          <i
            class="anticon anticon-down ant-select-arrow-icon"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </i>
        </span>
      </div>
    </div>
    <div
      class="ant-fullcalendar-month-select ant-select ant-select-enabled"
    >
      <div
        aria-autocomplete="list"
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-select-selection
            ant-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="ant-select-selection__rendered"
        >
          <div
            class="ant-select-selection-selected-value"
            style="display: block; opacity: 1;"
            title="Oct"
          >
            Oct
          </div>
        </div>
        <span
          class="ant-select-arrow"
          unselectable="on"
        >
          <i
            class="anticon anticon-down ant-select-arrow-icon"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </i>
        </span>
      </div>
    </div>
    <div
      class="ant-radio-group ant-radio-group-outline ant-radio-group-default"
    >
      <label
        class="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
      >
        <span
          class="ant-radio-button ant-radio-button-checked"
        >
          <input
            checked=""
            class="ant-radio-button-input"
            type="radio"
            value="date"
          />
          <span
            class="ant-radio-button-inner"
          />
        </span>
        <span>
          月
        </span>
      </label>
      <label
        class="ant-radio-button-wrapper"
      >
        <span
          class="ant-radio-button"
        >
          <input
            class="ant-radio-button-input"
            type="radio"
            value="month"
          />
          <span
            class="ant-radio-button-inner"
          />
        </span>
        <span>
          年
        </span>
      </label>
    </div>
  </div>
  <div
    class="ant-fullcalendar ant-fullcalendar-full ant-fullcalendar-fullscreen"
    tabindex="0"
  >
    <div
      class="ant-fullcalendar-calendar-body"
    >
      <table
        cellspacing="0"
        class="ant-fullcalendar-table"
        role="grid"
      >
        <thead>
          <tr
            role="row"
          >
            <th
              class="ant-fullcalendar-column-header"
              role="columnheader"
              title="Sun"
            >
              <span
                class="ant-fullcalendar-column-header-inner"
              >
                Su
              </span>
            </th>
            <th
              class="ant-fullcalendar-column-header"
              role="columnheader"
              title="Mon"
            >
              <span
                class="ant-fullcalendar-column-header-inner"
              >
                Mo
              </span>
            </th>
            <th
              class="ant-fullcalendar-column-header"
              role="columnheader"
              title="Tue"
            >
              <span
                class="ant-fullcalendar-column-header-inner"
              >
                Tu
              </span>
            </th>
            <th
              class="ant-fullcalendar-column-header"
              role="columnheader"
              title="Wed"
            >
              <span
                class="ant-fullcalendar-column-header-inner"
              >
                We
              </span>
            </th>
            <th
              class="ant-fullcalendar-column-header"
              role="columnheader"
              title="Thu"
            >
              <span
                class="ant-fullcalendar-column-header-inner"
              >
                Th
              </span>
            </th>
            <th
              class="ant-fullcalendar-column-header"
              role="columnheader"
              title="Fri"
            >
              <span
                class="ant-fullcalendar-column-header-inner"
              >
                Fr
              </span>
            </th>
            <th
              class="ant-fullcalendar-column-header"
              role="columnheader"
              title="Sat"
            >
              <span
                class="ant-fullcalendar-column-header-inner"
              >
                Sa
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          class="ant-fullcalendar-tbody"
        >
          <tr
            class=""
            role="row"
          >
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-last-month-cell ant-fullcalendar-last-day-of-month"
              role="gridcell"
              title="September 30, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  30
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 1, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  01
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 2, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  02
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 3, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  03
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 4, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  04
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 5, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  05
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 6, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  06
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
          </tr>
          <tr
            class=""
            role="row"
          >
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 7, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  07
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 8, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  08
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 9, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  09
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 10, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  10
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 11, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  11
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 12, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  12
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 13, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  13
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
          </tr>
          <tr
            class="ant-fullcalendar-current-week ant-fullcalendar-active-week"
            role="row"
          >
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 14, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  14
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 15, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  15
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 16, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  16
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 17, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  17
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 18, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  18
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-today ant-fullcalendar-selected-day"
              role="gridcell"
              title="October 19, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  19
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 20, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  20
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
          </tr>
          <tr
            class=""
            role="row"
          >
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 21, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  21
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 22, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  22
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 23, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  23
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 24, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  24
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 25, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  25
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 26, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  26
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 27, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  27
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
          </tr>
          <tr
            class=""
            role="row"
          >
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 28, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  28
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 29, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  29
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell"
              role="gridcell"
              title="October 30, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  30
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-last-day-of-month"
              role="gridcell"
              title="October 31, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  31
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-next-month-btn-day"
              role="gridcell"
              title="November 1, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  01
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-next-month-btn-day"
              role="gridcell"
              title="November 2, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  02
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-next-month-btn-day"
              role="gridcell"
              title="November 3, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  03
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
          </tr>
          <tr
            class=""
            role="row"
          >
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-next-month-btn-day"
              role="gridcell"
              title="November 4, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  04
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-next-month-btn-day"
              role="gridcell"
              title="November 5, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  05
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-next-month-btn-day"
              role="gridcell"
              title="November 6, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  06
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-next-month-btn-day"
              role="gridcell"
              title="November 7, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  07
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-next-month-btn-day"
              role="gridcell"
              title="November 8, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  08
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-next-month-btn-day"
              role="gridcell"
              title="November 9, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  09
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
            <td
              class="ant-fullcalendar-cell ant-fullcalendar-next-month-btn-day"
              role="gridcell"
              title="November 10, 2018"
            >
              <div
                class="ant-fullcalendar-date"
              >
                <div
                  class="ant-fullcalendar-value"
                >
                  10
                </div>
                <div
                  class="ant-fullcalendar-content"
                />
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
